styles/components/badges/styles.less

```diff
@@ -0,0 +1,450 @@
+& when (@badge-enabled) {
+  /* Badge */
+
+  .badge {
+    align-items: center;
+    display: inline-flex;
+    justify-content: center;
+    .element-shape-style(badge);
+    .element-spacing(badge, null, null);
+    .element-text-size(badge, null);
+    .element-text-style(badge);
+    .property-variant(badge, border-radius, null, null);
+
+    /* Layout Variants */
+
+    &.badge-narrow {
+      .element-spacing(badge, narrow, null);
+    }
+
+    &.badge-wide {
+      .element-spacing(badge, wide, null);
+    }
+
+    & > * {
+      display: inline-flex !important;
```

Comment:
Yes.  On line 6, the `.badge` is set to `display: inline-flex` with `align-items: center`.  However, if the children for whatever reason are defined as `display: block` or `display: inline` it can create complications -- from what I remember when testing.  Just tried it out without it and it seems to work fine.  So removing for now.